src/scripts/librarymenu.js

```diff
@@ -155,297 +302,671 @@ define(["dom", "layoutManager", "inputManager", "connectionManager", "events", "
+                        href: Dashboard.getConfigurationPageUrl(pluginItem.Name),
+                        pageUrls: [Dashboard.getConfigurationPageUrl(pluginItem.Name)]
+                    });
+                }
+            }
         }
     }
 
     function getToolsMenuLinks(apiClient) {
-        return apiClient.getJSON(apiClient.getUrl("web/configurationpages") + "?pageType=PluginConfiguration&EnableInMainMenu=true").then(createToolsMenuList, function(err) {
-            return createToolsMenuList([])
-        })
+        return apiClient.getJSON(apiClient.getUrl("web/configurationpages") + "?pageType=PluginConfiguration&EnableInMainMenu=true").then(createToolsMenuList, function (err) {
+            return createToolsMenuList([]);
+        });
     }
 
     function getToolsLinkHtml(item) {
-        var menuHtml = "",
-            pageIds = item.pageIds ? item.pageIds.join("|") : "";
+        var menuHtml = "";
+        var pageIds = item.pageIds ? item.pageIds.join("|") : "";
         pageIds = pageIds ? ' data-pageids="' + pageIds + '"' : "";
         var pageUrls = item.pageUrls ? item.pageUrls.join("|") : "";
-        return pageUrls = pageUrls ? ' data-pageurls="' + pageUrls + '"' : "", menuHtml += '<a is="emby-linkbutton" class="navMenuOption" href="' + item.href + '"' + pageIds + pageUrls + ">", item.icon && (menuHtml += '<i class="md-icon navMenuOptionIcon">' + item.icon + "</i>"), menuHtml += '<span class="navMenuOptionText">', menuHtml += item.name, menuHtml += "</span>", menuHtml += "</a>"
+        pageUrls = pageUrls ? ' data-pageurls="' + pageUrls + '"' : "";
+        menuHtml += '<a is="emby-linkbutton" class="navMenuOption" href="' + item.href + '"' + pageIds + pageUrls + ">";
+
+        if (item.icon) {
+            menuHtml += '<i class="md-icon navMenuOptionIcon">' + item.icon + "</i>";
+        }
+
+        menuHtml += '<span class="navMenuOptionText">';
+        menuHtml += item.name;
+        menuHtml += "</span>";
+        return menuHtml += "</a>";
     }
 
     function getToolsMenuHtml(apiClient) {
-        return getToolsMenuLinks(apiClient).then(function(items) {
-            var i, length, item, menuHtml = "";
-            for (menuHtml += '<div class="drawerContent">', i = 0, length = items.length; i < length; i++) item = items[i], item.href ? menuHtml += getToolsLinkHtml(item) : item.name && (menuHtml += '<h3 class="sidebarHeader">', menuHtml += item.name, menuHtml += "</h3>");
-            return menuHtml += "</div>"
-        })
+        return getToolsMenuLinks(apiClient).then(function (items) {
+            var i;
+            var length;
+            var item;
+            var menuHtml = "";
+
+            for (menuHtml += '<div class="drawerContent">', i = 0, length = items.length; i < length; i++) {
+                item = items[i];
+
+                if (item.href) {
+                    menuHtml += getToolsLinkHtml(item);
+                } else {
+                    if (item.name) {
```

Comment:
The if shouldn't be nested here.

Reply:
It isnt? Did you check the latest commit or one of the earlier ones? I broke it up into pieces, looking at my code this isn't actually in the last commit.

Reply:
GitHub had issues when I reviewed it I think. I was looking at the total changes 🤷‍♂️

Reply:
This is what I see
![image](https://user-images.githubusercontent.com/675118/51299029-34990100-1a27-11e9-9ebc-e80aa2f751fe.png)

It should be `else if` and not `else { if {` (forgive the formatting).